plugins/woocommerce/src/Admin/Features/LaunchYourStore.php

```diff
@@ -264,4 +270,54 @@ public function reset_woocommerce_coming_soon_banner_dismissed( $user_login, $us
 			WCAdminUser::update_user_data_field( $user->ID, self::BANNER_DISMISS_USER_META_KEY, 'no' );
 		}
 	}
+
+	/**
+	 * Check if the Mailpoet is connected.
+	 *
+	 * @return bool true if Mailpoet is fully connected, meaning the API key is valid and approved.
+	 */
+	private function is_mailpoet_connected() {
+		if ( ! class_exists( '\MailPoet\DI\ContainerWrapper' ) || ! class_exists( '\MailPoet\Settings\SettingsController' ) ) {
+			return false;
+		}
+
+		$container = \MailPoet\DI\ContainerWrapper::getInstance( WP_DEBUG );
+
+		// SettingControll retrieves data from wp_mailpoet_settings table.
+		$settings = $container->get( \MailPoet\Settings\SettingsController::class );
+
+		if ( false === $settings instanceof \MailPoet\Settings\SettingsController ) {
+			return false;
+		}
+
+		$mta       = $settings->get( 'mta' );
+		$api_state = $mta['mailpoet_api_key_state'] ?? null;
+
+		if ( ! $api_state || ! isset( $api_state['state'], $api_state['code'] ) ) {
+			return false;
+		}
+
+		return 'valid' === $api_state['state'] && 200 === $api_state['code'];
+	}
+
+	/**
+	 * Load slotfill script and JS variables for the newsletter.
+	 * The comingSoonNewsletter is used in client/wp-admin-scripts/coming-soon-newsletter-panel
+	 *
+	 * @return void
+	 */
+	public function load_newsletter_scripts() {
```

Comment:
Looks like we load the newsletter scripts in all admin pages. Should we only load it in site editor, especially when the coming soon template is loaded, if it's possible?

Reply:
Good call! Updated it in [Only include the script on site-editor page](https://github.com/woocommerce/woocommerce/pull/52650/commits/62d9e257f8b4552acda48aef7436b54e31736b77)